model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -76,21 +78,22 @@ public OrganizationModel create(String name, Set<String> domains) {
             throw new ModelDuplicateException("A organization with the same name already exists.");
         }
 
-        OrganizationEntity entity = new OrganizationEntity();
-        entity.setId(KeycloakModelUtils.generateId());
+        OrganizationAdapter adapter = new OrganizationAdapter(realm, this);
 
-        GroupModel group = createOrganizationGroup(entity.getId());
-
-        entity.setGroupId(group.getId());
-        entity.setRealmId(realm.getId());
-        entity.setName(name);
-        entity.setEnabled(true);
+        try {
+            session.setAttribute(OrganizationModel.class.getName(), adapter);
```

Comment:
We need this because the validations happen based on whether the `OrganizationModel` is not available in the session. If not, it means we are running outside the provider boundaries so we should fail any attempt to update/remove the group.

I'm also not very happy about this boilerplate code but otherwise, we will need to keep the previous changes where the validation happens in multiple places, something I think is not good from a maintenance perspective.

Also, the previous changes are missing places where groups are managed other than the Admin REST API. For instance, when running mappers or managing groups internally.

I don't have an alternative to achieve the same goal (have validations in a single place) so please let me know if you other suggestion.